packages/flutter/lib/src/material/text_field.dart

```diff
@@ -371,6 +381,38 @@ class TextField extends StatefulWidget {
   /// textfield's internal gesture detector, use a [Listener].
   final GestureTapCallback onTap;
 
+  /// Function to generate a custom counter widget. Passed the build context,
+  /// current and max lengths of the input, and [isFocused]. Will be placed
+  /// below the line in place of the default widget built when passing
+  /// [counterText].
+  ///
+  /// Accessibility will be up to you if you use this instead of [counterText].
```

Comment:
The returned widget should be accessible, just as the widget generated for [counterText] is.

Best not to refer to the reader as "you" (and etc)